src/App.tsx

```diff
@@ -1,18 +1,57 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/Errors/Errors';
+import { FilterType } from './types/FilterType';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState(FilterType.All);
+  const visibleTodos = useMemo<Todo[]>(() => {
+    if (todos) {
+      return [...todos].filter(todo => {
+        switch (filter) {
+          case FilterType.Active:
+            return !todo.completed;
+          case FilterType.Completed:
+            return todo.completed;
+          case FilterType.All:
+          default:
+            return true;
+        }
+      });
+    }
+
+    return [];
+  }, [todos, filter]);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+    setErrorMessage('');
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
-  }, []);
+
+    if (user) {
+      getTodos(user.id)
+        .then(result => setTodos(result))
+        .catch(() => {
+          setErrorMessage('Cannot load todos');
+        });
```

Comment:
I think it's better to use here try/catch construction and async/await syntax.

Reply:
agree, use async await and try catch